### ISSUE TITLE
CWL: default values specified in inputs section of subworkflow are ignored

### ISSUE THREAD

**Issue by KerstenBreuer** [DataBiosphere/toil#2502]
If I specify a default value for an input parameter in a subworkflow cwl file, this default value seems to be ignored.

Here is a simplified example:

main_workflow.cwl:
```
cwlVersion: v1.0
class: Workflow
requirements:
  SubworkflowFeatureRequirement: {}
inputs:
    bam:
        type: File
steps:
    sub_workflow:
        run: "./sub_workflow.cwl"
        in:
            bam:
                source: bam
        out:
            - bam_filt
outputs:
    bam_filt:
        type: File
        outputSource: sub_workflow/bam_filt
```

sub_workflow.cwl:
```
cwlVersion: v1.0
class: Workflow
inputs:
    bam:
        type: File
    is_paired_end:
        type: boolean
        default: true # this parameter is only specified in the subworkflow
steps:
    filtering:
        run: "./samtools_view.cwl"
        in:
            bam:
                source: bam
            is_paired_end:
                source: is_paired_end
        out:
            - bam_filt
outputs:
    bam_filt:
        type: File
        outputSource: filtering/bam_filt
```

samtools_view.cwl:
```
cwlVersion: v1.0
class: CommandLineTool
hints:
  DockerRequirement:
    dockerPull: kerstenbreuer/samtools:1.7

baseCommand: ["samtools", "view", "-h", "-b"]

inputs:
  bam:
    type: File
    inputBinding:
      position: 10
  is_paired_end:
    doc: if paired end, only properly paired reads pass
    type: boolean
    default: true
    inputBinding:
        prefix: "-f 3"
        shellQuote: false

arguments:
  - valueFrom: 4
    prefix: -F
    position: 1
  - valueFrom: 20
    prefix: -q
    position: 1

stdout: $(inputs.bam.nameroot + "_filt.bam")

outputs:
  bam_filt:
    type: stdout
```

Finnaly I get an error `WorkflowException: Missing required input parameter 'is_paired_end'`

I am working with toil v.3.17.0. When trying the same with cwltool, everythings fine.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-60)
┆Issue Number: TOIL-60

**Comment by ionox0** [comment]
Does it work when you try:
```
            is_paired_end: is_paired_end
```
instead of
```
            is_paired_end:
                source: is_paired_end
```

**Comment by DailyDreaming** [comment]
Stale, closing.  If this is still an issue, please reopen.

**Comment by mr-c** [comment]
I  can confirm this issues existed as recently as toil 3.24.0 and is fixed in toil 4.0.0; thank you @KerstenBreuer for the report and sorry for the lack of timely response
